lib/codegen/fromcto/jsonschema/jsonschemavisitor.js

```diff
@@ -411,6 +408,42 @@ class JSONSchemaVisitor {
                 jsonSchema.description = 'The instance identifier for this type';
             }
 
+        } else if (ModelUtil.isMap?.(field)) {
+            const mapDeclaration = field.getParent().getModelFile().getModelManager().getType(field.getFullyQualifiedTypeName());
+
+            let mapKey   = mapDeclaration.getModelFile().getType(mapDeclaration.getKey().getType());
+            let mapValue = mapDeclaration.getModelFile().getType(mapDeclaration.getValue().getType());
+
+            const jsonSchema = {
+                $id: field.getFullyQualifiedName(),
+                schema: {
+                    title: mapDeclaration.getName(),
+                    description : `An instance of ${field.getFullyQualifiedTypeName()}`,
+                    type: 'object',
+                    propertyNames: {
+                        type: 'string'
+                    },
+                    additionalProperties: {
+                        type: this.getFieldOrScalarDeclarationValidatorsForSchema(mapDeclaration.getValue().getType()).type
+                    }
+                }
+            };
+
+            if (mapKey.isScalarDeclaration?.() && mapKey.getValidator()?.getRegex()) {
+                jsonSchema.schema.propertyNames.pattern = String(mapKey.getValidator().getRegex());
```

Comment:
If we add support for regex validators for scalars we should also support it for string properties.

Reply:
@dselman this already exists! See [2f5b920](https://github.com/accordproject/concerto-codegen/pull/63/commits/2f5b9206afa8d350ba92b0f764e9f8bde2a718b8)

Reply:
That wasn't part of the original spec unfortunately. Thats a fundamental change at the parser level, and on through validation. We should add this as an enhancement in a follow up 

fyi @mttrbrts